### ISSUE TITLE
wrong bookmark encoding in PDF if using LuaLaTeX

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#2517]
Steps to reproduce create a Sphinx projet with file `index.rst`:

``` rst

Welcome to fixbookmarks's documentation!
========================================

If lualatex
-----------

wrong: éèùàï
------------

```

Do `make latexpdf PDFLATEX=lualatex`. Open the PDF in a viewer and open the bookmark panel:

![capture d ecran 2016-05-02 a 10 17 15](https://cloud.githubusercontent.com/assets/2589111/14950081/36c13914-104f-11e6-93a3-151f0c13fe7c.png)

(the issue is independent of project language English or French for example)

There is no issue with `make latexpdf PDFLATEX=xelatex` or with `make latexpdf`. The explanation is that `hyperref` package creates a `foo.out` file containing the bookmarks. For the Unicode engines, the file is in UTF-8. The `xelatex` then converts to `UTF-16BE` expected by PDF specification. The `lualatex` does no conversion when producing the PDF.

The fix would be to load `hyperref` with option `pdfencoding=unicode`. This tells `hyperref` to create the `foo.out` using `UTF-16BE` encoded bookmarks, except for `xetex` which hyperref treats especially as it is `xetex` which will do conversion from `UTF-8`. Thus the `pdfencoding=unicode` option works fine with `pdflatex`,  `xelatex`,  and `lualatex`.

However I see in `sphinx.sty` some special measures for Japanese

``` latex
\ifx\kanjiskip\undefined\else
  \usepackage{atbegshi}
  \ifx\ucs\undefined
    \ifnum 42146=\euc"A4A2
      \AtBeginShipoutFirst{\special{pdf:tounicode EUC-UCS2}}
    \else
      \AtBeginShipoutFirst{\special{pdf:tounicode 90ms-RKSJ-UCS2}}
    \fi
  \else
    \AtBeginShipoutFirst{\special{pdf:tounicode UTF8-UCS2}}
  \fi
\fi
```

Unfortunately I am ignorant of Japanese, hence I can not test whether this could or not be removed in favor of `pdfencoding=unicode` option given to `hyperref`.

**Comment by jfbu** [comment]
In fact I don't know what the `\special{pdf:tounicode ...}` does and whether it really concerns bookmarks. At any rate one could already do:

``` latex
\ifx\kanjiskip\undefined\PassOptionsToPackage{pdfencoding=unicode}{hyperref}\else
[as above]
```

Or one could perhaps use `pdfencoding=unicode` option always: but I don't know to test `make latexpdfja` ...

**Comment by jfbu** [comment]
Notice for clarity that the `pdfencoding` option to `hyperref` only concerns the encoding used for the bookmarks. There are three possibilities: `pdfdoc` which works only for a subset of latin1 encoding (244 characters), `unicode` which uses `UTF-16BE` (except that the job is left to  `xetex` if used), and `auto` which uses the `pdfdoc` encoding when the character string contains only characters allowing it (use less bytes) , else uses  `UTF-16BE`.

**Comment by jfbu** [comment]
Fixed at 0f0d5058185ce252961ce672d3697655fbde691c
